cdc/processor/pipeline/sink.go

```diff
@@ -340,28 +354,38 @@ func (n *sinkNode) HandleMessage(ctx context.Context, msg pipeline.Message) (boo
 			failpoint.Inject("ProcessorSyncResolvedError", func() {
 				failpoint.Return(false, errors.New("processor sync resolved injected error"))
 			})
-			if err := n.flushSink(ctx, msg.PolymorphicEvent.CRTs); err != nil {
+			err := n.retryFlushSink(ctx, event.CRTs)
+			if err != nil {
 				return false, errors.Trace(err)
 			}
-			atomic.StoreUint64(&n.resolvedTs, msg.PolymorphicEvent.CRTs)
+			atomic.StoreUint64(&n.resolvedTs, event.CRTs)
 			return true, nil
 		}
-		if err := n.emitEvent(ctx, event); err != nil {
+		err := n.addEvent2Buffer(event)
+		if err != nil {
 			return false, errors.Trace(err)
 		}
+
+		var ok bool
+		for !ok {
+			ok, err = n.tryEmitRow2Sink(ctx)
+			if err != nil {
+				return false, errors.Trace(err)
+			}
+		}
```

Comment:
Looks like it is a busy loop (which should be avoided) when `tryEmitRow2Sink` keeps returning false.